src/run/logs/DockerComposeWatchLogHandler.ts

```diff
@@ -3,96 +3,94 @@ import { LogHandler, Logger } from "./logs-handler.js";
 import stripAnsi from "strip-ansi";
 
 /**
- * Handles log entries for development mode, parsing and logging
+ * Handles log entries for development watch logs, parsing and logging
  * specific events such as service ready, healthy, restart, and crash.
  */
 export class DevelopmentWatchLogNodeHandler implements LogHandler {
-    private static readonly RESTART_REGEX = /"?([\w-]+)"?\s+\|\s+.*Nodemon Restarting.../;
-    private static readonly CRASHED_REGEX = /"?([\w-]+)"?\s+\|\s+.*Nodemon Crashed!/;
-    private static readonly READY_REGEX = /"?([\w-]+)"?\s+\|\s+.*Application Ready\./;
-    private static readonly BUILT_STATUS_REGEX = /"?([\w-]+)"?\s+.*exited with code 0/;
-    private static readonly NPM_INSTALL_COMPLETE_REGEX = /"?([\w-]+)"?\s+\|\s+.*npm install commencing\./;
-    private static readonly NPM_INSTALL_FAILED_REGEX = /"?([\w-]+)"?\s+\|\s+.*npm install failed!\./;
+    // Regular expressions for matching log patterns
+    private static readonly LOG_PATTERNS = {
+        RESTART: /"?([\w-]+)"?\s+\|\s+.*Nodemon Restarting.../,
+        CRASHED: /"?([\w-]+)"?\s+\|\s+.*Nodemon Crashed!/,
+        READY: /"?([\w-]+)"?\s+\|\s+.*Application Ready\./,
+        HEALTHY_STATUS: /Container\s+([\w-]+)\s+healthy/,
+        UNHEALTHY_STATUS: /Container\s+([\w-]+)\s+unhealthy/,
+        NPM_INSTALL_COMPLETE: /"?([\w-]+)"?\s+\|\s+.*npm install commencing\./,
+        NPM_INSTALL_FAILED: /"?([\w-]+)"?\s+\|\s+.*npm install failed!\./
+    };
```

Comment:
Looks to only handle NPM/Node actions - is there a plan to handle java in this manner too?

Reply:
Java services tracks the health status of the container using the health check api. For other non-builders, that will be handled separately - perhaps when we move the go and perl builder outside of chs-dev.